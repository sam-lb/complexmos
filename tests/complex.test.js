

const { complex, Complex } = require("../math/complex.js");
const snapShotDiff = require("snapshot-diff");

expect.addSnapshotSerializer(snapShotDiff.getSnapshotDiffSerializer());


test("1 + 1 = 2", () => {
    expect(complex(1, 0).add(complex(1, 0)).re).toBe(2);
    expect(complex(1, 0).add(complex(1, 0)).im).toBe(0);
});


test("test complex addition", () => {
    expect(complex(1.12, 43.5).add(complex(51.309, 12.2))).toMatchSnapshot();
});