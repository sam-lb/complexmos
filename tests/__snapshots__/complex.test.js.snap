// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test complex addition 1`] = `
Complex {
  "im": 55.7,
  "re": 52.428999999999995,
}
`;
